front/lib/api/assistant/conversation.ts

```diff
@@ -630,16 +570,167 @@ export async function* retryAgentMessage(
   | AgentGenerationSuccessEvent
   | AgentMessageSuccessEvent
 > {
-  yield {
-    type: "agent_error",
-    created: Date.now(),
-    configurationId: "foo",
-    messageId: "bar",
-    error: {
-      code: "not_implemented",
-      message: "Not implemented",
+  const {
```

Comment:
I think we can easily move all the event emission out of the transaction

It returns stuff or null if null that just means we haven't found the message to retry otherwise we can move the rendering code and event creation code out of the transaction making the "in-transaction" code much shorter

But I can also live with your current implementation so feel free to change or not